jscomp/others/js.ml

```diff
@@ -246,6 +246,9 @@ module String2 = Js_string2
 module Re = Js_re
 (** Provide bindings to JS regex expression *)
 
+module React = Js_react
```

Comment:
Not sure if I like this design. Previously the user had at least at least the chance to manipulate the `React` bindings namespace (`React.createElement`, etc).

If I had a `Jsx` module that complies to a certain type / function / value structure, I could at least provide my own framework that uses that `Jsx.something` facilities. Am I missing something obvious here? @cristianoc

Reply:
As an example:

```res
let foo = <MyFoo test="test"/>
```

could potentially expand to

```res
let foo = Jsx._jsx(\"MyFoo\", {test: "test"})
```

where I had a module `Jsx` defined like this...

```res
// Jsx.res
external jsx : 'props component -> 'props -> element = "jsx"
  [@@bs.module "react/jsx-runtime"]
```

The actual common types would need to be derived from a common file (kinda like we do with `Dom`), so frameworks can align on the same tokens?

(Sorry for my vagueness, the examples may be technically wrong, but hopefully you get my point)

Reply:
> As an example:
> 
> ```rescript
> let foo = <MyFoo test="test"/>
> ```
> 
> could potentially expand to
> 
> ```rescript
> let foo = Jsx._jsx(\"MyFoo\", {test: "test"})
> ```
> 
> where I had a module `Jsx` defined like this...
> 
> ```rescript
> // Jsx.res
> external jsx : 'props component -> 'props -> element = "jsx"
>   [@@bs.module "react/jsx-runtime"]
> ```
> 
> The actual common types would need to be derived from a common file (kinda like we do with `Dom`), so frameworks can align on the same tokens?
> 
> (Sorry for my vagueness, the examples may be technically wrong, but hopefully you get my point)

I totally, can't agree more. Here is my idea aligned to your comment https://github.com/rescript-lang/rescript-react/pull/49#issuecomment-1172852091